cupy/padding/pad.py

```diff
@@ -1,16 +1,98 @@
 import cupy
+import six
 
 
-def pad(array, pad_width, mode=None, **kwargs):
-    if not isinstance(pad_width, int):
-        raise NotImplementedError
+def _prepend_const(arr, pad_amt, val, axis=-1):
+    if pad_amt == 0:
+        return arr
+    padshape = tuple(x if i != axis else pad_amt
+                     for (i, x) in enumerate(arr.shape))
+    if val == 0:
+        return cupy.concatenate((cupy.zeros(padshape, dtype=arr.dtype), arr),
+                                axis=axis)
+    else:
+        return cupy.concatenate(((cupy.zeros(padshape) + val).astype(
+                                 arr.dtype), arr), axis=axis)
+
+
+def _append_const(arr, pad_amt, val, axis=-1):
+    if pad_amt == 0:
+        return arr
+    padshape = tuple(x if i != axis else pad_amt
+                     for (i, x) in enumerate(arr.shape))
+    if val == 0:
+        return cupy.concatenate((arr, cupy.zeros(padshape, dtype=arr.dtype)),
+                                axis=axis)
+    else:
+        return cupy.concatenate(
+            (arr, (cupy.zeros(padshape) + val).astype(arr.dtype)), axis=axis)
+
+
+def _normalize_shape(ndarray, shape, cast_to_int=True):
+    ndims = ndarray.ndim
+    if shape is None:
+        return ((None, None), ) * ndims
+    arr = cupy.asarray(shape)
```

Comment:
Why do you use `cupy.ndarray` for `shape`? Is it really efficient? I think `numpy.ndarray` is enough.